pkg/scheduler/objects/application.go

```diff
@@ -53,6 +53,11 @@ const (
 	AppTagStateAwareDisable string = "application.stateaware.disable"
 )
 
+type EventLogEntry struct {
+	EventTime        time.Time
+	ApplicationState string
+}
```

Comment:
This should be called StateLogEntry, we log state changes not every event.

Reply:
Updated in new commit.